examples/ecs/observers.rs

```diff
@@ -16,18 +16,18 @@ fn main() {
         // Observers are systems that run when an event is "triggered". This observer runs whenever
```

Comment:
You've mentioned wanting to avoid repeated churn / breaking changes here. Which I like! One major element of churn that I expect to see even after this PR is relations-based spawning of observers.

I currently feel that:

- we badly need this: simply punting on the problem for more cycles is not viable as BSN / UI work will make this a serious blocker soon
- this PR / series of PRs should not implement those changes
- we should not block Bevy 0.17's release on this. While they are related breaking changes, they're not "the exact same code" in a way that's very annoying
- this PR should make those changes much easier to make: the internals are much simpler

Reply:
Agreed!